sync-todo/v2/client/cpp/controllers/app_controller.cpp

```diff
@@ -0,0 +1,64 @@
+#include "app_controller.hpp"
+
+void AppController::onFrame() {
+  _navigation.onFrame();
+}
+
+AppController::AppController(ftxui::ScreenInteractive *screen) {
+  _appState.screen = screen;
+
+  /** Read the contents of the atlasConfig.json to get the metadata for the App Services App.
+   * This path assumes you are running the app from a `/build` directory within this project. If you're
+   * running the app somewhere else, update the path to your atlasConfig.json accordingly. */
+  std::ifstream f("../atlasConfig.json");
+  nlohmann::json data = nlohmann::json::parse(f);
+  auto appConfigMetadata = data.template get<AppConfigMetadata>();
+  f.close();
```

Comment:
Added taking the path as a command line argument. Could you explain more about why I would want to load the file async, though? I would assume we _want_ to read the file synchronously because we require information that's in the file to initialize the App.

Reply:
It's not advised to do any blocking procedures on the UI thread. If the file system is unavailable, this call would block the UI and cause the OS to want to kill the app.